### PR TITLE
Add IsUsageBased to features and send it to web UI

### PR DESCRIPTION
**Pull request by mcbattirola**
This PR adds a `isUsageBased` flag to the `Features` struct and sends it to the web UI.

This flag will be used to hide/show billing-related stuff in the web UI. Currently, the flag is not used.

This flag is set by this PR: https://github.com/gravitational/teleport.e/pull/1260

### CODE REVIEW

**Review by michellescripts — APPROVED** [review]

File: lib/modules/modules.go [hunk 1]
```diff
@@ -66,6 +66,8 @@ type Features struct {
 	Plugins bool
 	// AutomaticUpgrades enables automatic upgrades of agents/services.
 	AutomaticUpgrades bool
+	// IsUsageBased enables some usage-based billing features
+	IsUsageBased bool
```
Comment (kimlisa) [inline lib/modules/modules.go:70]: to keep the flag name consistent and more descriptive?
```suggestion
	// IsUsageBasedBilling enables some usage-based billing features
	IsUsageBasedBilling bool
```

File: api/client/webclient/webconfig.go [hunk 2]
```diff
@@ -61,6 +61,8 @@ type WebConfig struct {
 	// The web UI for dashboards provides functionality for downloading self-hosted licenses and
 	// Teleport Enterprise binaries.
 	IsDashboard bool `json:"isDashboard,omitempty"`
+	// IsUsageBasedBilling determines if the user subscription is usage-based (pay-as-you-go).
```
Comment (kimlisa) [inline api/client/webclient/webconfig.go:64]: to make it clear it's a cloud related flag?
```suggestion
	// IsUsageBasedBilling determines if the cloud user subscription is usage-based (pay-as-you-go).
```

File: web/packages/teleport/src/config.ts [hunk 3]
```diff
@@ -40,6 +40,7 @@ const cfg = {
   isDashboard: false,
   tunnelPublicAddress: '',
   recoveryCodesEnabled: false,
+  isUsageBasedBilling: false,
```
Comment (kimlisa) [inline web/packages/teleport/src/config.ts:43]: can we copy over the comment describing this flag? (we are working on adding more helpful comments in the web UI)

**Review by kimlisa — APPROVED** [review]

**Review by timothyb89 — APPROVED** [review]
LGTM with @kimlisa's comments resolved.

**Pull request merged**